cpp/dolfinx/fem/utils.cpp

```diff
@@ -133,6 +133,55 @@ fem::DofMap fem::create_dofmap(
   return DofMap(layout, index_map, bs, std::move(dofmaps.front()), bs);
 }
 //-----------------------------------------------------------------------------
+std::vector<fem::DofMap> fem::create_dofmaps(
+    MPI_Comm comm, const std::vector<ElementDofLayout>& layouts,
+    mesh::Topology& topology,
+    std::function<void(const std::span<std::int32_t>&, std::uint32_t)>
+        unpermute_dofs,
+    std::function<std::vector<int>(const graph::AdjacencyList<std::int32_t>&)>
+        reorder_fn)
+{
+  // Create required mesh entities
+  const int D = topology.dim();
+  for (int d = 0; d < D; ++d)
+  {
+    if (layouts.front().num_entity_dofs(d) > 0)
+      topology.create_entities(d);
+  }
+
+  auto [_index_map, bs, dofmaps]
+      = build_dofmap_data(comm, topology, layouts, reorder_fn);
+  auto index_map = std::make_shared<common::IndexMap>(std::move(_index_map));
+
+  // If the element's DOF transformations are permutations, permute the
+  // DOF numbering on each cell
+  if (unpermute_dofs)
+  {
+    if (layouts.size() != 1)
+    {
+      throw std::runtime_error(
+          "DOF transformations not yet supported in mixed topology.");
+    }
+    const int D = topology.dim();
+    const int num_cells = topology.connectivity(D, 0)->num_nodes();
```

Comment:
`int` type?

Let's be very careful with `int` types. We've taken a lot of care to get it right.